lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2537,6 +2537,13 @@ namespace {
           cxxRecordDecl->getName() == "pair") {
         forceMemberwiseInitializer = true;
       }
+      // If this is the type that wraps around a Swift closure for the purpose
+      // of std::function support, force a memberwise initializer. It will be
+      // called by the synthesized std::function initializer.
+      if (cxxRecordDecl && cxxRecordDecl->getIdentifier() &&
```

Comment:
I see that we already have this practice but I wonder if it is even possible to have a record decl that does not have an identifier. So I wonder if these checks are redundant. (It is possible to have some decls that do not have a single identifiers, like overloaded operators, but I wonder if this never applies to records unless there is an extension I am not aware of.)